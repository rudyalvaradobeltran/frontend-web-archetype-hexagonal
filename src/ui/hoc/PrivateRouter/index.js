import React from 'react';
import { Route } from 'react-router-dom';

const PrivateRouter = ({ children, ...rest }) => (
  <Route
    {...rest}
    render={() =>
      localStorage.getItem('token') ? (
        children
      ) : (
        <React.Fragment>{window.location.replace('/')}</React.Fragment>
      )
    }
  />
);

export default PrivateRouter;
