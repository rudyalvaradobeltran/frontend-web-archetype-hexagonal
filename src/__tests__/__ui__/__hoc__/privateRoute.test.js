import React from 'react';
import { create } from 'react-test-renderer';

// Mocks
import MockProvider from '../../../__mocks__/mockProvider';
// Components
import PrivateRouter from '../../../ui/hoc/PrivateRouter';

describe('PrivateRouter', () => {
  const Component = () => <div>home</div>;

  it('Should renders correctly (Snapshot)', () => {
    const wrapper = create(
      <MockProvider initialEntries="/home">
        <PrivateRouter path="/home">
          <Component />
        </PrivateRouter>
      </MockProvider>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('Should renders correctly (Snapshot) children when token is present in localStorage', () => {
    window.localStorage.setItem('token', 'secret-token');
    
    const wrapper = create(
      <MockProvider initialEntries="/home">
        <PrivateRouter path="/home">
          <Component />
        </PrivateRouter>
      </MockProvider>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
