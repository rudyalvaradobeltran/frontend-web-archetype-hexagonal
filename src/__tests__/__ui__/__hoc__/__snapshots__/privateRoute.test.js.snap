// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRouter Should renders correctly (Snapshot) 1`] = `null`;

exports[`PrivateRouter Should renders correctly (Snapshot) children when token is present in localStorage 1`] = `
<div>
  home
</div>
`;

exports[`PrivateRouter Should renders correctly (Snapshot) with icons 1`] = `null`;
